src/app/shared/components/loader/loader.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, NgZone, Output, EventEmitter } from '@angular/core';
+
+@Component({
+  selector: 'app-loader',
+  templateUrl: './loader.component.html',
+})
+export class LoaderComponent {
+  progress = 0;
+
+  @Output()
+  progressFinished: EventEmitter<void> = new EventEmitter<void>();
+
+  constructor(private ngZone: NgZone) {
```

Comment:
Can you add some documentation here (maybe at the top of this class or in a separate readme.md in this folder)?

What do you want to showcase here? Why is NgZone needed for this? Where can I find additional info? 
These questions should be answered with some short sentences in the code so that in the future we know why this is here.